src/components/Card/index.jsx

```diff
@@ -1,27 +1,88 @@
 import React, { Component } from 'react';
 import './index.css';
 
+import {BsPencil, BsX, BsCheck} from 'react-icons/bs';
+
 class Card extends Component {
+
     state = {
-        isChecked: false
+        isChecked: false,
+        isEdit: false,
+        cardData : {
+            cardLogo: 'Caption',
+            text: 'Some text here...',
+        },
+        tempCardData: {
+            tempLogo: 'Caption',
+            tempText: 'Some text here...'
+        }
     }
+    
 
     changeStyles = () => {
         this.setState({isChecked : !this.state.isChecked});
     }
+    changeEditStatus = () => {
+        this.setState({isEdit : !this.state.isEdit, isChecked: false});
+    }
+    handleChangeLogo = (event) => {
+        console.log('piu');
+        this.setState({tempCardData: {tempLogo: event.target.value, tempText: this.state.tempCardData.tempText}});
+    }
+    handleChangeText = (event) => {
+        console.log('piu-piu');
+        this.setState({tempCardData: {tempText: event.target.value, tempLogo: this.state.tempCardData.tempLogo}});
+    }
+    changeCardInfo = () => {
+        console.log('piu-piu-piu');
+        this.setState({cardData: {cardLogo: this.state.tempCardData.tempLogo, text: this.state.tempCardData.tempText}, isChecked: false});
+        this.changeEditStatus();
+    }
     
     render() {
         return (
             <div className="Card">
                 <div className="textMarg">
                     <div className={this.state.isChecked ? "green" : null}>
-                        <label className="bold">Caption</label>
-                        <input type="checkbox" className="topRight" onClick={this.changeStyles} />
+                        <div className="headCard">
+                            {this.state.isEdit ? 
```

Comment:
You have 3 logical ternary blocks checking this.state.isEdit. FAR more better to unite them and have one ternary block and check it once..